--- libcaca-0.99.beta18/ruby/caca-display.c.orig	2012-04-06 22:36:17.000000000 +0200
+++ libcaca-0.99.beta18/ruby/caca-display.c	2012-12-04 20:28:37.435444821 +0100
@@ -84,7 +84,7 @@
 
     if(display == NULL)
     {
-        rb_raise(rb_eRuntimeError, strerror(errno));
+        rb_raise(rb_eRuntimeError, "%s", strerror(errno));
     }
 
     _SELF = display;
@@ -131,7 +131,7 @@
 {
     if(caca_set_display_title(_SELF, StringValuePtr(t))<0)
     {
-        rb_raise(rb_eRuntimeError, strerror(errno));
+        rb_raise(rb_eRuntimeError, "%s", strerror(errno));
     }
     return t;
 }
@@ -252,7 +252,7 @@
 {
     if(caca_set_display_driver(_SELF, StringValuePtr(driver))<0)
     {
-        rb_raise(rb_eRuntimeError, strerror(errno));
+        rb_raise(rb_eRuntimeError, "%s", strerror(errno));
     }
     return driver;
 }
@@ -267,7 +267,7 @@
 {
     if(caca_set_cursor(_SELF, flag)<0)
     {
-        rb_raise(rb_eRuntimeError, strerror(errno));
+        rb_raise(rb_eRuntimeError, "%s", strerror(errno));
     }
     return flag;
 }
