--- libcaca-0.99.beta2/cxx/caca++.cpp.orig	2006-04-27 10:22:08.000000000 +0200
+++ libcaca-0.99.beta2/cxx/caca++.cpp	2006-07-02 23:10:30.671123000 +0200
@@ -47,7 +47,7 @@
 
 void Caca::setDelay(unsigned int d)
 {
-    caca_set_delay(dp, d);
+    caca_set_display_time(dp, d);
 }
 
 void Caca::Display()
@@ -57,7 +57,7 @@
 
 unsigned int Caca::getRendertime()
 {
-    return caca_get_rendertime(dp);
+    return caca_get_display_time(dp);
 }
 
 unsigned int Caca::getWidth()
